src/utils/validators.ts

```diff
@@ -1,5 +1,7 @@
-export function assertIsHexString(value: any): asserts value is `0x${string}` {
-  if (typeof value !== 'string' || !/^0x[0-9a-fA-F]+$/.test(value)) {
-    throw new Error('Value must be of type `0x${string}`');
+import { isAddress } from 'viem';
+
+export function assertIsAddress(value: any): asserts value is `0x${string}` {
```

Comment:
@douglance I've updated `assertIsHexString` to `assertIsAddress` because just checking if something's a hex string isn't enough, as the value should be a valid Ethereum address. I used `isAddress` from `viem` for the check.

Reply:
I'm cool with that, I'll make the change.